queries.sql

```diff
@@ -8,4 +8,10 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.50;
 SELECT * FROM animals WHERE neutered = true;
 SELECT * FROM animals WHERE name != 'Gabumon';
 SELECT* FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+SELECT COUNT(*) FROM animals;
+SELECT COUNT(escape_attempts) FROM animals WHERE escape_attempts = 0;
+SELECT AVG(weight_kg) FROM animals;
+SELECT neutered, COUNT(escape_attempts) FROM animals GROUP BY neutered ORDER BY COUNT(escape_attempts) DESC LIMIT 1;
+SELECT species MAX(weight_kg), MIN(weight_kg) FROM animals GROUP BY species;
+SELECT species, AVG(escape_attempts) FROM animals WHERE date_of_birth BETWEEN DATE '1990-01-01' AND '2000-01-01' GROUP BY species;
```

Comment:
#### queries.sql

1. Students should create queries to answer the following questions in transactions. Please note: if students find another way that presented in the example that also works, it is fine!
   * Inside a transaction:
     * `UPDATE` the animals table by setting the species column to unspecified, then roll back the change (\*\*for this step students are explicitly asked to verify that change was made and rollbacked, so make sure that they included all steps as in the example below \*\*).

     ```sql
        BEGIN; -- start transaction
     
        UPDATE animals
        SET species = 'unspecified'; -- make change
     
        SELECT species from animals; -- verify that change was made
        ROLLBACK;
     
        SELECT species from animals; -- verify that change was undone
     ```
   * Inside a transaction:
     * `UPDATE` the animals table by setting the species column to digimon for all animals that have a name ending in mon.
     * `UPDATE` the animals table by setting the species column to pokemon for all animals that don't have species already set.
     * Commit the transaction.

     ```sql
        BEGIN;
        UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon'; 
        UPDATE animals SET species = 'pokemon' WHERE species IS NULL; 
     
        SELECT species from animals; -- verify that change was made
        COMMIT;
        SELECT species from animals; -- verify that change persists after commit
     ```
   * Inside a transaction delete all records in the animals table, then roll back the transaction.

     ```sql
        BEGIN;
     
        DELETE FROM animals WHERE ;
        SELECT COUNT(*) FROM ANIMALS;
     
        ROLLBACK;
        SELECT COUNT(*) FROM ANIMALS;
     
     ```
   * Inside a transaction:
     * `DELETE` all animals born after Jan 1st, 2022
     * Create a savepoint for the transaction
     * `UPDATE` all animals' weight to be their weight multiplied by -1.
     * Rollback to the savepoint
     * `UPDATE` all animals' weights that are negative to be their weight multiplied by -1.
     * Commit transaction

     ```sql
         BEGIN;
     
         DELETE FROM animals WHERE  date_of_birth > '2022/01/01' ;
         SAVEPOINT SP1;
     
         UPDATE animals SET weight_kg = weight_kg * -1; 
         ROLLBACK TO SP1;
         UPDATE animals SET weight_kg = weight_kg * -1 WHERE weight_kg < 0; 
     
         COMMIT;
     ```